icu4c/source/common/locid.cpp

```diff
@@ -1357,7 +1361,9 @@ Locale::setKeywordValue(const char* keywordName, const char* keywordValue, UErro
         }
         fullName = newFullName;
         status = U_ZERO_ERROR;
-        newLength = uloc_setKeywordValue(keywordName, keywordValue, fullName, newLength, &status) + 1;
+        uloc_setKeywordValue(keywordName, keywordValue, fullName, newLength, &status) + 1;
```

Comment:
The final `+ 1` here makes no sense at all, this value is never used for anything.